From 07daa825f4a4af9e269ace4071b86e6528412619 Mon Sep 17 00:00:00 2001
From: Rob Browning <rlb@defaultvalue.org>
Date: Sun, 10 Apr 2011 10:03:45 -0500
Subject: Expect 0 from kill() for zombies on GNU/Hurd.

Emacs should no longer hang while waiting for child processes on GNU/Hurd.

Author: Samuel Thibault <sthibault@debian.org>
Date: Mon, 31 Jan 2011 01:13:35 +0100
Applied-By: Rob Browning <rlb@defaultvalue.org>
Forwarded: bug-gnu-emacs@gnu.org
Closes: #611591
---
 src/sysdep.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/sysdep.c b/src/sysdep.c
index 1fbc061..3abb43f 100644
--- a/src/sysdep.c
+++ b/src/sysdep.c
@@ -411,7 +411,7 @@ wait_for_termination (pid)
   while (1)
     {
 #ifdef subprocesses
-#if defined (BSD_SYSTEM) || defined (HPUX)
+#if (defined (BSD_SYSTEM) || defined (HPUX)) && !defined(__GNU__)
       /* Note that kill returns -1 even if the process is just a zombie now.
 	 But inevitably a SIGCHLD interrupt should be generated
 	 and child_sig will do wait3 and make the process go away. */
